packages/overlay/src/integrations/sentry/index.ts

```diff
@@ -103,34 +103,51 @@ export default function sentryIntegration(options?: SentryIntegrationOptions) {
   } satisfies Integration<Envelope>;
 }
 
+/**
+ * Implements parser for
+ * @see https://develop.sentry.dev/sdk/envelopes/#serialization-format
+ * @param rawEvent Envelope data
+ * @returns parsed envelope
+ */
 export function processEnvelope(rawEvent: RawEventContext) {
   const { data } = rawEvent;
-  const [rawHeader, ...rawEntries] = data.trim().split(/\n/gm);
-
-  const header = JSON.parse(rawHeader) as Envelope[0];
-  const items: Envelope[1][] = [];
-  for (let i = 0; i < rawEntries.length; i += 2) {
-    // guard both rawEntries[i] and rawEntries[i + 1] are defined and not empty
-    if (!rawEntries[i] || !rawEntries[i + 1]) {
-      continue;
+  let prevCursor = 0;
+  let cursor = data.indexOf('\n');
```

Comment:
The last newline in an envelope is always optional, which includes an empty envelope with just its header. This means you should default to to the end of the string here. 

Right now, `indexOf` will return `-1` and the slice below will skip the last character (presumably a `}`), which will fail parsing. 

Test with `const data = '{}'`, which should succeed and give you a completely empty envelope.

In items below you handle this correctly.